lib/Sema/DerivedConformanceDifferentiable.cpp

```diff
@@ -388,10 +388,7 @@ deriveBodyDifferentiable_zeroTangentVectorInitializer(
                               DeclNameLoc(), /*Implicit*/ true);
 
     // Create closure expression.
-    DiscriminatorFinder DF;
-    for (Decl *D : parentDC->getParentSourceFile()->getTopLevelDecls())
-      D->walk(DF);
-    auto discriminator = DF.getNextDiscriminator();
+    unsigned discriminator = 0;
```

Comment:
I suppose there's no chance of mangled name conflict due to closure discriminators here because synthesized `zeroTangentVectorInitializer` declarations contain only one closure.

Reply:
Correct. The DiscriminatorFinder is actually wrong because it assumed that discriminators were globally unique, which is why you saw them incrementing in your test case when really parsed closures would've all had the discriminator of 0 in this code.